### ISSUE TITLE
Unable to NFS share on Mounted Virtual Drive Using mirror exe.

### ISSUE THREAD

**Issue by DHARANIDHARAN993** [dokan-dev/dokany#496]
# Environment
 * Windows version: 2012 R2 Server
 * Dokany version: 0.8

# Description
I have mounted a virtual drive using mirror exe. But am not able to NFS Share on the mounted virtual Drive. In Version 0.6,  I have changed the file System to NTFS. After that, NFS sharing option is showed in the properties but am not able to do NFS share. At the Same in version 0.8, After changing the filesystem, NFS sharing options in not showing in the properties. 

Added to this, After Mounting, tried to NFS share using cmd and powershell command is also not working. 

Thanks in adavance

**Comment by Liryna** [comment]
Hi @DHARANIDHARAN993 ,

Dokany 0.8 is not the last release, can you try with 1.0.3 https://github.com/dokan-dev/dokany/releases/tag/v1.0.3 ?
You should not have issue to do a NFS share.

**Comment by DHARANIDHARAN993** [comment]
I have installed the latest version of  [dokan]( https://github.com/dokan-dev/dokany/releases/tag/v1.0.3) and tried the same. NFS sharing options is not showing in properties. And Also i tried using Powershell Command as the same. No luck. 

Dokan Version: 
`Driver path: 'C:\Windows\system32\drivers\dokan1.sys'`
`dokanctl : Mar 24 2017 14:25:11`
`Dokan version : 100`
`Dokan driver version : 0x190`

Command Used to Mount Virtual Drive : `mirror.exe /r C:\dokany-0.6.0  /l Z`

Properties of the created Drive And doesn't Contain NFS sharing Option. 

Instruct me if I did anything wrong.
Thanks

![dokan](https://cloud.githubusercontent.com/assets/16775855/25169836/19045024-2506-11e7-8e81-7cbba2d004de.PNG)

**Comment by Liryna** [comment]
@DHARANIDHARAN993 OK now I see what you want to do sorry.
I have not personally did this but looks like someone succeed in doing it. See https://github.com/dokan-dev/dokany/issues/466

**Comment by Daniel-Abrecht** [comment]
No, I didn't actually try to do that. What I mentioned in #466 refered to my Port of Ronnie Sahlbergs' fuse-nfs program: https://github.com/Daniel-Abrecht/fuse-nfs-crossbuild-scripts/
I used the dokany-fuse wrapper to port it so windows users don't need an expensive windows enterprise version just to mount an NFS share. It doesn't help in any way to create an NFS share in windows.

**Comment by Liryna** [comment]
Thank you @Daniel-Abrecht for you answer !
Therefor, it is probably Windows that do not display the NFS option for a reason.
@DHARANIDHARAN993 can you try to mount with `/o` option and see if it change something ?
Otherwise we will need mirror & kernel logs here.

**Comment by DHARANIDHARAN993** [comment]
Thank you @Liryna for your answer !
FYI ..
I have tried the older version say 0.6. After mounting,the virtually mounted drive file system is `DOKAN`. Then I changed the file system from `DOKAN` to `NTFS`. After this, NFS sharing option is showed in Drive properties but NFS sharing is not working. Then I tried the same with later version,i cant find NFS sharing option.
So i suspect that there is no reason to hide NFS option from windows. I think we missing some properties in Mirror source Code.
And Also i tried mounting with option `/o`. Result is same.I cant find any mirror logs while executing mirror. 
My Requirement is `Need to do NFS share on Virtually Mounted Drive`. Any Other Alternate way is also welcome.

**Comment by Rondom** [comment]
Hm, so let's see what is different.
Can you produce mirror logs when opening the explorer dialog (both where it works and where it does not work)? I am not sure if it is needed, but maybe you can also provide a log from [ProcessMonitor](https://technet.microsoft.com/en-us/sysinternals/processmonitor).

**Comment by DHARANIDHARAN993** [comment]
I have attached the Mirror logs and Kernel Logs.
[Mirror.txt](https://github.com/dokan-dev/dokany/files/946803/Mirror.txt)
[KernalMirrorLog.txt](https://github.com/dokan-dev/dokany/files/946804/KernalMirrorLog.txt)

**Comment by Rondom** [comment]
I could reproduce "NFS Sharing" not showing up for Dokan drives under Win2016 with Dokan 1.0.3 and the snapshot 1.0.3.1172.
I tried with unmodified Dokan 0.6.0 (reporting as DOKAN FS-type) and indeed it did not work with this one either (as you described). Can you share your modified version?

I tried getting logs with ProcessMonitor and I succeeded for the newer Dokany releases. Dokan 0.6.0 has some bug that makes all paths in ProcessMonitor be `explorer.exe` instead of the real path. This bug was only fixed in later Dokan versions. 
Easiest to produce the logs in ProcessMonitor is, if one checks "run folder windows in a separate process" in the Explorer options. This makes it easier to filter only for the explorer window of interest in Process Monitor.
Attached is the ProcessMonitor log of Dokan103 and NTFS (invoking properties using Alt-Enter on "cygdrive"-folder).

[explorer_properties_nfs_cygdrive.zip](https://github.com/dokan-dev/dokany/files/949102/explorer_properties_nfs_cygdrive.zip)

**Comment by A4Abinash** [comment]
Actually the issue is with the Windows NFS Server itself. Server for NFS works only for physical folders, thus for virtual drive folders the NFS sharing does not work. 

Though i tried creating a mirror between a physical drive and already mounted virtual drive which failed. Is there any way we can provide NFS sharing for virtual drives created using DOKAN.
